### ISSUE TITLE
[Suggestion] Add a Publish Book Event

### ISSUE THREAD

**Issue by Starza4TW** [SpongePowered/SpongeAPI#1313]
Add an event so that plugins can detect when a player clicks the "Sign and Close" button that turns a Book and Quill into a Written Book

**Comment by parlough** [comment]
Good idea, but does this need its own event? Could this simply fire the `AffectItemStackEvent`? If the original is a book and quill and the replacement is a written book in the transaction you know it has been signed. 

I'd be willing to implement either of them so just looking for some feedback here.

**Comment by ryantheleach** [comment]
AffectItemStackEvent would be a little generic, I think that having support where possible for specific vanilla events is ok for ease of use. This event would probably extend AffectItemStackEvent.

https://github.com/SpongePowered/SpongeAPI/issues/743 (closed) Suggests that it may already exist as an 'inventory event' , But I'll leave this open with new comments as it's been a few years and people are still requesting this.
